### PR TITLE
docs: add readme and examples for js client

### PR DESCRIPTION
**Pull request by prakharmathur82**
* Add README
* Add example usages

### CODE REVIEW

File: clients/js/examples/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (punit-kulal) [inline clients/js/examples/package.json:1]: Do we need it to be a separate package ? The go client [examples] (https://github.com/raystack/raccoon/tree/main/clients/go/examples) do not define it as a separate package. Any reason to have a separate package.

File: clients/js/.eslintignore [hunk 2]
```diff
@@ -1 +1,2 @@
 protos
+examples
```
Comment (punit-kulal) [inline clients/js/.eslintignore:2]: Any reason to keep them out of eslint and prettier

File: clients/js/README.md [hunk 3]
```diff
@@ -1 +1,181 @@
...
+
+- Type: `Optional`
+- Default value: `{}`
+
+#### `timeout`
+
+The request timeout in milliseconds.
+
+- Type: `Optional`
+- Default value: `5000`
```
Comment (punit-kulal) [inline clients/js/README.md:120]: Should we keep the default timeout as 1000, since that's the default in other clients.

**Review by punit-kulal — CHANGES_REQUESTED** [review]

File: clients/js/examples/package.json (hunk 1 above)
Comment (prakharmathur82) [reply clients/js/examples/package.json:1]: wanted to include raystack/raccoon from npm package

File: clients/js/.eslintignore (hunk 2 above)
Comment (prakharmathur82) [reply clients/js/.eslintignore:2]: mostly because it's test code

File: clients/js/README.md (hunk 3 above)
Comment (prakharmathur82) [reply clients/js/README.md:120]: ok

File: clients/js/examples/package.json (hunk 1 above)
Comment (punit-kulal) [reply clients/js/examples/package.json:1]: Understood

File: clients/js/.eslintignore (hunk 2 above)
Comment (punit-kulal) [reply clients/js/.eslintignore:2]: Even tests are under formatter and lint if I am not wrong. Would be preferrable to have everything under the formatter.

Comment (prakharmathur82) [reply clients/js/.eslintignore:2]: ack

File: clients/js/README.md [hunk 4]
```diff
@@ -1 +1,181 @@
...
+    .catch(error => {
+        console.error('Error:', error);
+    });
+```
+
+**Note:** Here, the PageEvent and ClickEvent are imported from generated protobufjs code.
+
+See complete examples for sending JSON and protobuf messages: [examples](examples)
+
+## Configurations
```
Comment (punit-kulal) [inline clients/js/README.md:85]: Won't a tabular format be easier to read and follow. The line by line on readme, looks a bit sparse.

File: clients/js/README.md [hunk 5]
```diff
@@ -1 +1,181 @@
...
+   ```
+4. Run link.
+
+   ```sh
+   $ npm run lint
+   ```
+
+## Versioning
+
+We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags](https://github.com/raystack/raccoon/tags).
```
Comment (punit-kulal) [inline clients/js/README.md:181]: Shouldn't this point to raccoon's npm package instead. Since, not all raccoon tags have a js client.

**Comment by punit-kulal** [comment]
You would also want to add a files field in your client's package.json or add a .npmignore an make sure the examples repository is excluded. Otherwise it will also be included in your build/dist.

File: clients/js/README.md (hunk 4 above)
Comment (prakharmathur82) [reply clients/js/README.md:85]: I am following the usual config format that in being followed in raytstack

example: https://github.com/raystack/raccoon/blob/main/docs/docs/reference/configurations.md

File: clients/js/README.md (hunk 5 above)
Comment (prakharmathur82) [reply clients/js/README.md:181]: ya actually past tags won't be there, will change

File: clients/js/README.md (hunk 3 above)
Comment (prakharmathur82) [reply clients/js/README.md]: done

File: clients/js/.eslintignore (hunk 2 above)
Comment (prakharmathur82) [reply clients/js/.eslintignore]: done

**Review by punit-kulal — APPROVED** [review]

**Pull request merged**